app/models/rook.rb

```diff
@@ -1,3 +1,8 @@
 class Rook < Piece
 
+    def is_valid?(new_x, new_y)
+    	return false unless super
+    	(new_x == self.x_coordinate || new_y == self.y_coordinate)? true : false

```

Comment:
No need for the ternary operator. The check will already return `true` Or `false`.

Reply:
thanks @ilyakrasnov; not really 100% sure what ternary operator means but I think I understand you in so far as (new_x == self.x_coordinate || new_y == self.y_coordinate) will return true if either of these conditions is met. Tested and working.

Reply:
ternary operator is what you have used here: `condition ? do sth : do sth else`. Like `if else` on one line.. Ternary because it has 3 parts :)

we could also think about giving the two conditions a name/concept and extracting in a separate method.. so that the code would look like:
```
def ...
  return false unless super
  moving_vertically?(new_x) || moving_horizontally?(new_y)
end 
```

This way you can skim through the code and understand the valid move instantly.. if you want to dive deeper and see how its implemented, you go to the corresponding methods and find the details there.. Hope it makes sense...

Reply:
okay got you - it would be more lines of code but more supportable. On the case. ;)
RE: the tests and vertical move to row 1 or 8, as per Yoshi's call we manually removed the pawn at row 2 or 7 in test case to allow a move to row 1 or 8. A bit of a hack for now but allows tests that you can move vertically to all valid parts of the board.